src/client_side.cc

```diff
@@ -657,28 +656,26 @@ clientSetKeepaliveFlag(ClientHttpRequest * http)
 }
 
 /// checks body length of non-chunked requests
-static int
+static bool
 clientIsContentLengthValid(HttpRequest * r)
 {
     // No Content-Length means this request just has no body, but conflicting
     // Content-Lengths mean a message framing error (RFC 7230 Section 3.3.3 #4).
     if (r->header.conflictingContentLength())
-        return 0;
-
-    switch (r->method.id()) {
-
-    case Http::METHOD_GET:
+        return false;
 
-    case Http::METHOD_HEAD:
-        /* We do not want to see a request entity on GET/HEAD requests */
-        return (r->content_length <= 0 || Config.onoff.request_entities);
+    // RFC 7230 section 3.3 - Content-Length defines body exists regardless of method
+    if (r->content_length > 0)
```

Comment:
If we should exclude zero Content-Length, we should explain why it should be excluded. If it should be excluded, perhaps the name of the function (and its description) should be changed.

Reply:
Good point. Will do that on next update.

Reply:
Done in f2a7292